usr/lib/zypp/plugins/urlresolver/susecloud

```diff
@@ -48,6 +63,39 @@ class SUSECloudPlugin(Plugin):
                 credentials = headers.get('credentials')
                 if credentials:
                     repo_url += '?credentials=' + credentials
+                    if verify_credentials:
+                        credentials_file_path = (
+                            '/etc/zypp/credentials.d/%s' % credentials
+                        )
+                        user, password = utils.get_credentials(
+                            credentials_file_path
+                        )
+                        if not user or not password:
+                            self.error(
+                                {'CREDENTIAL_ERROR': 'NOCREDENTIALS'},
+                                ('Credentials required for "%s" but not found '
+                                 'in "%s"' % (repo_url, credentials_file_path))
+                            )
+                            return
+                        # Verify access to the server withthe credentials
+                        auth_creds = HTTPBasicAuth(user, password)
```

Comment:
Move everything related to checking the credentials also into a separate method, perhaps?